sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -146,6 +146,14 @@ namespace Azure { namespace Core {
      */
     Context() : m_contextSharedState(std::make_shared<ContextSharedState>()) {}
 
+    /**
+     * @brief Copy-constructs a new context from another `%Context` instance.
+     *
+     * @param other Another instance to copy.
+     *
+     */
+    Context(const Context& other) = default;
```

Comment:
@CaseyCarter: If we need a copy ctor and copy assignment, should we add an explicit `default` destructor, to follow the 0-3-5 rule (and get to three)?
https://en.cppreference.com/w/cpp/language/rule_of_three

Or, is the better alternative to remove the copy assignment operator, since it is defaulted anyway and go to **zero**?

Reply:
FWIW, this item is currently also included in #3954 since I also ran into the exact same issue.

Reply:
I would remove the explicitly-defaulted copy assignment (the only effect of which was to keep the compiler from defining a copy constructor) instead of adding an explicitly-defaulted copy constructor. There will still be a defaulted copy assignment - now implicit instead of explicit - and there will be a new implicit copy constructor as well. I greatly prefer 0 to 3 or 5 when it works, because it's the simplest/default behavior.

Reply:
Would removing the explicitly defaulted copy assignment be considered a source breaking change? Or will it be opaque to most callers?

Reply:
> Would removing the explicitly defaulted copy assignment be considered a source breaking change? Or will it be opaque to most callers?

100% non-breaking. This class will be every bit as copy-assignable as before, I'd expect all codegen for existing programs to be bit-for-bit identical.

Reply:
FWIW, I made Casey's suggested fix to the #3954 pull request and it worked like a champ.